aws-predictions/src/main/java/com/amplifyframework/predictions/aws/AWSPredictionsPlugin.java

```diff
@@ -79,7 +91,26 @@ public String getPluginKey() {
     @Override
     public void configure(JSONObject pluginConfiguration, @NonNull Context context) throws PredictionsException {
         this.configuration = AWSPredictionsPluginConfiguration.fromJson(pluginConfiguration);
-        this.predictionsService = new AWSPredictionsService(configuration);
+
+        AWSCredentialsProvider credentialsProvider;
+
+        if (credentialsProviderOverride != null) {
+            credentialsProvider = credentialsProviderOverride;
+        } else {
+            try {
+                credentialsProvider =
+                        (AWSMobileClient) Amplify.Auth.getPlugin(AUTH_DEPENDENCY_PLUGIN_KEY).getEscapeHatch();
+            } catch (IllegalStateException exception) {
+                throw new PredictionsException(
+                        "AWSPredictionsPlugin depends on AWSCognitoAuthPlugin but it is currently missing",
+                        exception,
+                        "Before configuring Amplify, be sure to add AWSPredictionsPlugin same as you added " +
+                                "AWSPinpointAnalyticsPlugin."
```

Comment:
Copypasta.

Reply:
Yeah gonna clean this all up after GA - it's all throughout the framework.